### PR TITLE
[PBL-473] [PBL-516] Subsquid: batch processor and failed Pablo transactions

### PR DESCRIPTION
**Pull request by andresmechali**
https://app.clickup.com/t/20465559/PBL-473
https://app.clickup.com/t/20465559/PBL-516

**MAYBE** fix https://app.clickup.com/t/20465559/PBL-515

### CODE REVIEW

**Comment by KaiserKarel** [comment]
Task linked: [PBL-516 Subsquid: stored failed transactions](https://app.clickup.com/t/20465559/PBL-516)

**Comment by KaiserKarel** [comment]
Task linked: [PBL-473 Subsquid - Implement batch processing](https://app.clickup.com/t/20465559/PBL-473)

**Review by easteregg — APPROVED** [review]

**Review by fl-y — APPROVED** [review]

File: subsquid/src/server-extension/resolvers/assets.ts [hunk 1]
```diff
@@ -37,7 +37,19 @@ export class AssetsResolver {
 
     const manager = await this.tx();
 
-    const price = await getOrCreateHistoricalAssetPrice(manager, assetId, new Date(date || new Date()).getTime());
+    if (!date) {
+      const currentPrices = await getCurrentAssetPrices(manager);
+      const price =
+        currentPrices?.[assetId] ||
+        (await getOrCreateHistoricalAssetPrice(manager, assetId, new Date(date || new Date()).getTime()));
```
Comment (bengalmozzi) [inline subsquid/src/server-extension/resolvers/assets.ts:44]: date is null. just new Date().getTime() will be fine

File: subsquid/src/processorTypes.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  .addEvent("BondedFinance.OfferCancelled", {
+    data: { event: { extrinsic: { signature: true, hash: true }, args: true } }
+  } as const)
+  .addEvent("Vesting.VestingScheduleAdded", {
+    data: { event: { extrinsic: { signature: true, hash: true }, args: true } }
+  } as const)
+  .addEvent("Vesting.Claimed", {
+    data: { event: { extrinsic: { signature: true, hash: true }, args: true } }
+  } as const)
+  .addCall("Pablo.add_liquidity", {
```
Comment (bengalmozzi) [inline subsquid/src/processorTypes.ts:59]: The name of this event is unclear; What about naming it something like LiquidityAddedError? Same for Pablo.remove_liquidity, Pablo.swap

**Review by bengalmozzi — COMMENTED** [review]
Looks good sir, just small comments :)

File: subsquid/src/processorTypes.ts (hunk 2 above)
Comment (andresmechali) [reply subsquid/src/processorTypes.ts:59]: This is autogenerated, and comes from the chain extrinsics (https://github.com/ComposableFi/composable/blob/3b22417b82f39127449fc80b315c29664c0d9283/code/parachain/frame/pablo/src/lib.rs#L387)

File: subsquid/src/server-extension/resolvers/assets.ts (hunk 1 above)
Comment (andresmechali) [reply subsquid/src/server-extension/resolvers/assets.ts:44]: well spotted, carried it from a copy-paste 🙂

**Review by bengalmozzi — APPROVED** [review]
👍

**Pull request merged**